File: app/src/main/java/com/lazysoul/kotlinwithandroid/KotlinWithAndroid.java
package com.lazysoul.kotlinwithandroid;

import com.lazysoul.kotlinwithandroid.injection.components.ApplicationComponent;
import com.lazysoul.kotlinwithandroid.injection.components.DaggerApplicationComponent;
import com.lazysoul.kotlinwithandroid.injection.module.ApplicationModule;

import android.app.Application;

/**
 * Created by Lazysoul on 2017. 7. 15..
 */

public class KotlinWithAndroid extends Application {

    protected ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        applicationComponent = DaggerApplicationComponent
            .builder()
            .applicationModule(new ApplicationModule(this))
            .build();
    }

    public ApplicationComponent getComponent() {
        return applicationComponent;
    }

}

File: app/src/main/kotlin/com/lazysoul/kotlinwithandroid/KotlinWithAndroid.kt
package com.lazysoul.kotlinwithandroid

import android.app.Application
import com.lazysoul.kotlinwithandroid.injection.components.ApplicationComponent
import com.lazysoul.kotlinwithandroid.injection.components.DaggerApplicationComponent
import com.lazysoul.kotlinwithandroid.injection.module.ApplicationModule

/**
 * Created by Lazysoul on 2017. 7. 15..
 */

class KotlinWithAndroid : Application() {

    lateinit var component: ApplicationComponent

    override fun onCreate() {
        super.onCreate()

        component = DaggerApplicationComponent
            .builder()
            .applicationModule(ApplicationModule(this))
            .build()
    }

}
